src/Core/Content/Media/DataAbstractionLayer/MediaIndexer.php

```diff
@@ -97,16 +105,72 @@ public function handle(EntityIndexingMessage $message): void
             return;
         }
 
+        $context = $message->getContext();
```

Comment:
You should check in the `update` method the payload if you need to update urls. This should not run like when you assign a media to a product

Reply:
Why? :-) ThumbnailsRo is updated in the handler, too.
Additionally, there is a check within the set*Paths, what has to be updated.